internal/serve/httphandler/wallet_creation_handler.go

```diff
@@ -107,3 +114,30 @@ func (h WalletCreationHandler) GetWallet(rw http.ResponseWriter, req *http.Reque
 	}
 	httpjson.Render(rw, resp, httpjson.JSON)
 }
+
+func (h WalletCreationHandler) GetWalletStatus(rw http.ResponseWriter, req *http.Request) {
+	ctx := req.Context()
+	token := strings.TrimSpace(chi.URLParam(req, "token"))
+	if len(token) == 0 {
+		httperror.BadRequest("Token is required", nil, nil).Render(rw)
+		return
+	}
+
+	wallet, err := h.EmbeddedWalletService.GetWalletByToken(ctx, token)
+	if err != nil {
+		if errors.Is(err, services.ErrInvalidToken) {
+			httperror.NotFound("Wallet not found", err, nil).Render(rw)
+		} else {
+			httperror.InternalError(ctx, "Failed to get wallet status", err, nil).Render(rw)
+		}
+		return
+	}
+
```

Comment:
Is the token something used for authentication or for identification?

If for authentication, we should probably return 401 in both cases: (i) missing token; and (ii) `services.ErrInvalidToken`. The rationale being that any information we give to someone trying to brute force the application would make it easier for it to be brute forced.